carDB/src/main/java/pl/agh/edu/cardatabase/carCategory/ice/CarServant.java

```diff
@@ -2,24 +2,30 @@
 
 import com.zeroc.Ice.Current;
 import com.zeroc.Ice.ObjectNotExistException;
-import org.springframework.beans.factory.annotation.Autowired;
 import pl.agh.edu.cardatabase.car.persistence.CarRepository;
-import pl.agh.edu.cardatabase.carCategory.persistence.CarCategoryRepository;
 
-import java.util.NoSuchElementException;
+import java.util.Optional;
 
 public class CarServant implements cardb.Car {
-    @Autowired
-    private CarCategoryRepository carCategories;
-    @Autowired
-    private CarRepository cars;
+    private final CarRepository cars;
+
+    public CarServant(CarRepository carRepository) {
+        this.cars = carRepository;
+    }
 
     @Override
     public int getCategory(Current current) {
         try {
             int carId = Integer.parseInt(current.id.name);
-            return carCategories.getCarCategoryById(carId).get().getId();
-        } catch (Exception e) {
+            Optional<Integer> maybeCarCategoryId = cars.getCarCategoryId(carId);
```

Comment:
I think naming something "maybe..." is not a good naming convention, the if(isPresent()) part shows, that this may be empty, no need forr the "maybe" in the name